## Big Data Analysis

#### Preparation
Load the data from the notebooks

In [14]:
%run customers_zip.ipynb
%run customer_orders.ipynb
%run orders.ipynb
%run sports.ipynb
%run customer_zip_orders.ipynb

#### 1. How many customers practice more than 2 sports?

In [15]:
df_sports_cleaned['customer_id'].value_counts()[n_o_sports > 2].count()

np.int64(5040)

There are 5040 clients who practice more than two sports. The detailed analysis is in the file sports.ipynb

#### 2. Which sport is the least popular among customers?

In [16]:
df_sports_cleaned['sport'].value_counts(sort=True, ascending=True)

sport
hokej           21409
chodzenie       21423
snowboard       21472
judo            21476
aquafitness     21566
karate          21695
rowery          43038
bieganie        53111
jezdziectwo     63640
myslistwo      191489
nurkowanie     211743
lucznictwo     222829
plywanie       274855
Name: count, dtype: int64

From the table containing information about the occurrences of sport type in the data set it can be concluded that hockey (hokej) is the least popular sport. The detailed analysis is in the file sports.ipynb

#### 3. What is the average order value?

There are occurencies of orders with negative values, they were removed from the data set as I accounted them as returns from the customers. There were orders which had different id's but identical order_id's and values, I removed the duplicates.

In [17]:
df_orders_cleaned.describe()

order_id         value
count  3.964073e+06  3.964073e+06
mean   5.503874e+07  2.565577e+04
std    2.589908e+07  4.066933e+05
min    1.018763e+07  1.000000e+01
25%    3.261849e+07  2.094000e+02
50%    5.501393e+07  4.090400e+02
75%    7.746140e+07  6.085100e+02
max    9.991874e+07  8.000000e+06

The table containing the statistics of the orders in the data set shows that the average order value is 25655,77. The detailed analysis is in the file orders.ipynb


#### 4. From which postal code do customers most frequently make purchases, and what is the average value of their orders?

##### There are rare occurencies of clients having two postal codes. I calculated the most popular postal codes in two ways:
- the client has two houses - I used both postal codes
- the client moved over the years - I used the newest postal code


In [18]:
df_customers_multiple_zip_cleaned['zip_cust'].value_counts()

zip_cust
12-091    323
92-241    313
22-661    311
51-350    307
02-190    305
         ... 
30-000    207
60-021    207
21-610    206
42-670    205
00-650    191
Name: count, Length: 4200, dtype: int64

In [19]:
df_customers_one_zip_cleaned['zip_cust'].value_counts()

zip_cust
12-091    321
92-241    312
22-661    311
51-350    306
90-331    305
         ... 
60-021    206
30-000    206
21-610    205
42-670    205
00-650    191
Name: count, Length: 4200, dtype: int64

Regardless of whether we consider that the client has multiple homes or had moved, the most clients are from postal code 12-091. The detailed analysis is in the file customers_zip.ipynb

In [20]:
df_customer_order_values_most_popular_zip.describe()

customer_id      order_id         value
count  1.221000e+03  1.221000e+03  1.221000e+03
mean   4.286482e+07  5.667211e+07  4.951942e+04
std    2.699547e+07  2.656229e+07  5.726021e+05
min    6.431100e+04  1.030328e+07  1.174000e+01
25%    1.929414e+07  3.341342e+07  2.231400e+02
50%    4.198518e+07  5.791053e+07  4.130900e+02
75%    6.740764e+07  7.931701e+07  6.053600e+02
max    8.913473e+07  9.986605e+07  7.612700e+06

The table containing the statistics of the orders from customers with postal code 12-091 shows that the average order value is 49519,42. The detailed analysis is in the file customer_zip_orders.ipynb
